solar_system_wave3.rb

```diff
@@ -0,0 +1,261 @@
+  def planets_by_name(name)
+    planet_names = []
+    @planets.each do |planet_class|
+      planet_names << planet_class.name
+    end
+    index = planet_names.index(name.capitalize.to_s)
+    return @planets[index]
+  end
+
+# method to create an array one method for all planet classes
+  def all_planets(method)
+    method_array =[]
+    @planets.each do |planet_class|
+      method_array << planet_method(planet_class.name, method)
+    end
+    return method_array
+  end
+
+
+# method to print the planet names in various ways
+  def list(type)
+    planet_list = ""
+
+    all_planets(:name).length.times do |i|
+      planet_list += "#{i + 1}. #{all_planets(:name)[i]}"
+      if type == :line
+        planet_list += "\n"
+      elsif type == :tab
+        planet_list += "\t"
+      elsif type == :comma
+        planet_list += ", "
+      else
+        puts "error"
+      end
+    end
+
+    return planet_list
+  end
+
+# method to create a table with info from each planet called by the array passed to table_create
+  def table_create(table_arr)
+
+    planet_table_headings = ["Name", "|Mass", "|Index from Sun", "|Circumference", "|Color", "|Year Length", "|Distance from the Sun"]
+
+    planet_table_units = [" ", "|in kg", "|", "|in miles", "| ", "|in Earth years", "|in AU"]
+
+    printf "%-15s%-15s%-15s%-15s%-15s%-15s%-15s\n" % planet_table_headings
+    printf "%-15s%-15s%-15s%-15s%-15s%-15s%-15s\n" % planet_table_units
+
+    table_arr.length.times do |i|
+      planet_table_values = planet_method(table_arr[i], :all_facts)
+      puts "-" * 105
+      printf "%-15s|%-14s|%-14s|%-14s|%-14s|%-14s|%-14s\n" % planet_table_values
+    end
+  end
+
+# method to find the distance between planets
+  def distance_from_other(planet1, planet2)
+    distance = planets_by_name(planet2).distance_from_sun - planets_by_name(planet1).distance_from_sun
+    return distance.abs
+  end
+
+# method to find the amount of planet's years since the SS began
+  def years_since_birth(planet)
+    local_year = age / planets_by_name(planet).year_length
+    # local_year = "%e" %local_year #converts to scientific notation
+    return local_year
+  end
+
+end
+
+
+class Planet
+  attr_accessor :name, :mass, :index_from_sun, :circumference, :color, :year_length, :distance_from_sun
+
+  def initialize(name, mass, index_from_sun, circumference, color, year_length, distance_from_sun)
+    @name = name
+    @mass = mass
+    @index_from_sun = index_from_sun
+    @circumference = circumference
+    @color = color
+    @year_length = year_length
+    @distance_from_sun = distance_from_sun
+  end
+
+
+  def all_facts
+    info = [name, mass, index_from_sun, circumference, color, year_length, distance_from_sun]
+    return info
+  end
+
+end
+
+
+mercury_class = Planet.new("Mercury", 3.285e+23, 1, 9525, "Orange", 0.24, 0.39)
+venus_class = Planet.new("Venus", 4.87e+24, 2, 23628, "Yellow", 0.62, 0.72)
```

Comment:
You use the term `class` often in variable names. I would recommend avoiding this - in general, it's usually clear when you've got an instance of a class. Instead, calling these `mercury` or `venus` might be more readable.